chat-server/src/main/java/pl/vyraj/InputHandler.java

```diff
@@ -0,0 +1,115 @@
+package pl.vyraj;
+
+import java.io.BufferedWriter;
+import java.io.BufferedReader;
+import java.io.OutputStreamWriter;
+import java.io.InputStreamReader;
+import java.io.IOException;
+import java.net.Socket;
+import java.util.ArrayList;
+import java.util.List;
+
+public class InputHandler implements Runnable{
+    protected static final List<InputHandler> INPUT_HANDLERS = new ArrayList<>();
+    protected static final List<String> USER_NAMES = new ArrayList<>();
```

Comment:
Powinieneś tu użyć mimo wszystko kolekcji bezpieczenych pod względem wielowątkowości - nawet jeśli w danym momencie nie obsługujesz rejestracji wielopoziomowych to wciąż może to się zmienić w przyszłości.

Reply:
Now I'm using thread safe collection: CopyOnWriteArrayList.